crates/consensus/consensus/src/lib.rs

```diff
@@ -10,14 +10,24 @@
 
 use reth_primitives::{
     BlockHash, BlockNumber, BlockWithSenders, Bloom, GotExpected, GotExpectedBoxed, Header,
-    HeaderValidationError, InvalidTransactionError, Receipt, SealedBlock, SealedHeader, B256, U256,
+    HeaderValidationError, InvalidTransactionError, Receipt, Request, SealedBlock, SealedHeader,
+    B256, U256,
 };
 use std::fmt::Debug;
 
 #[cfg(any(test, feature = "test-utils"))]
 /// test helpers for mocking consensus
 pub mod test_utils;
 
+/// Post execution input passed to [Consensus::validate_block_post_execution].
+#[derive(Debug)]
+pub struct PostExecutionInput<'a> {
+    /// Receipts of the block.
+    pub receipts: &'a [Receipt],
+    /// EIP-7685 requests of the block.
+    pub requests: &'a [Request],
```

Comment:
I wonder if we should do Option here?

Reply:
we return an empty vector from block execution both when hardfork is enabled and when no requests are found. So I think it's easier to leave it as a vector here